test/sharness/t0040-add-and-cat.sh

```diff
@@ -9,6 +9,15 @@ test_description="Test add and cat commands"
 . lib/test-lib.sh
 
 test_add_cat_file() {
+	test_expect_success "ipfs add --help works" '
+		ipfs add --help > add_help_out
+	'
+
+	test_expect_success "stdin reading message doesnt show up" '
+		test_expect_code 1 grep "ipfs: Reading from" add_help_out &&
+		test_expect_code 1 grep "send Ctrl-d to stop." add_help_out
+	'
+
```

Comment:
This test would be successful even before the patch, you have to capture both stdout and stderr.